mypy/build.py

```diff
@@ -2172,6 +2171,16 @@ def finish_passes(self) -> None:
                                 typemap=self.type_map())
             manager.report_file(self.tree, self.type_map(), self.options)
 
+            self.update_fine_grained_deps(self.manager.fg_deps)
+            self.free_state()
+            if not manager.options.fine_grained_incremental and not manager.options.preserve_asts:
+                free_tree(self.tree)
```

Comment:
Is there a reason why we can't do `self.tree = None`? Or maybe doing `self.tree.defs.clear()` or something?

Reply:
The reason we need to go deeper than just clearing `defs` is that the symbol table contains references to functions and classes and the like, and we need to clear out their bodies.

It would be possible to drive this from the symbol table, and I considered that, but it is more complex for not much gain. (Probably a bit faster, but probably doesn't matter)

(We actually probably should also do `self.tree.defs.clear()`, which would free a little bit more memory, though probably not a substantial amount.)